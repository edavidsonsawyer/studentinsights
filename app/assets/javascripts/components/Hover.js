import PropTypes from 'prop-types';
import React from 'react';
import _ from 'lodash';


// A component for tracking mouse hover state
export default class Hover extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isHovering: false
    };

    this.onMouseLeave = this.onMouseLeave.bind(this);
    this.onMouseEnter = this.onMouseEnter.bind(this);
  }

  onMouseEnter(e) {
    this.setState({isHovering: true});
  }

  onMouseLeave(e) {
    this.setState({isHovering: false});
  }

  render() {
    const {children, className, style} = this.props;
    const {isHovering} = this.state;

    const classNameText = _.compact(['Hover', className]).join(' ');
    return (
      <div
        className={classNameText}
        style={style}
        onMouseLeave={this.onMouseLeave}
        onMouseEnter={this.onMouseEnter}>{children(isHovering)}</div>
    );
  }
}
Hover.propTypes = {
  children: PropTypes.func.isRequired,
  style: PropTypes.object,
  className: PropTypes.string
};