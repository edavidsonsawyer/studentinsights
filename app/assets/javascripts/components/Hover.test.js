import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Hover from './Hover';

function testProps(props) {
  return {
    children: isHovering => `some text, rendered based on isHovering: ${isHovering}`,
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<Hover {...testProps()} />, el);
  expect(el.textContent).toContain('some text, rendered based on isHovering: false');
});

it('snapshots view', () => {
  const tree = renderer
    .create(<Hover {...testProps()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});