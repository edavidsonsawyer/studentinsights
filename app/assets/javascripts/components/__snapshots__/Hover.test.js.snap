// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  className="Hover"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  some text, rendered based on isHovering: false
</div>
`;
